molecule/default/vars/Debian.yml

```diff
@@ -0,0 +1,2 @@
+---
+python_version: "3.11"
```

Comment:
What will be our mechanism for keeping this version number up-to-date for each platform?

Reply:
It would be better to grab the Python version from the system instead of hardcoding it.  That should be possible with, say, [`ansible.builtin.package_facts`](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/package_facts_module.html).

Reply:
Please let me know if https://github.com/cisagov/skeleton-ansible-role/pull/190/commits/fc22b2361b32b1776322fac59c4a0ce9b46038f2 works for both of you.

Reply:
That works for me.  If it works for you @jsf9k, please resolve this conversation.

Reply:
The code looks fine to me, but I don't understand why the `EXTERNALLY-MANAGED` file doesn't appear to get replaced [here](https://github.com/cisagov/skeleton-ansible-role/actions/runs/8912634940/job/24476479925?pr=190) or [here](https://github.com/cisagov/skeleton-ansible-role/actions/runs/8912634857/job/24476480927?pr=190).

What do you think about changing the dummy Molecule test to check that the system python is externally managed?  That would seem to be a worthwhile thing to do here.

Reply:
So I see `changed` for Debian 12 and `ok` for Debian 13. This implies the file was created for Debian 12 but already exists for Debian 13. I can confirm this because after running `molecule create` I get the following:

```console
$ docker container exec debian12-systemd ls -l /usr/lib/python3.11/EXTERNALLY-MANAGED
-rw-r--r-- 1 root root 645 May  7 19:28 /usr/lib/python3.11/EXTERNALLY-MANAGED
$ docker container exec debian13-systemd ls -l /usr/lib/python3.11/EXTERNALLY-MANAGED
-rw-r--r-- 1 root root 645 Apr 10 13:16 /usr/lib/python3.11/EXTERNALLY-MANAGED
```

The timestamp on the Debian 13 file indicates that it already existed. If I switch the order in the `prepare.yml` file so that the `EXTERNALLY-MANAGED` file is restored _before_ packages are upgraded I see the following:

```log
TASK [Restore EXTERNALLY-MANAGED file for Python] ******************************
skipping: [amazonlinux2023-systemd]
skipping: [debian10-systemd]
skipping: [debian11-systemd]
skipping: [fedora38-systemd]
skipping: [fedora39-systemd]
skipping: [kali-systemd]
skipping: [ubuntu-20-systemd]
skipping: [ubuntu-22-systemd]
changed: [debian12-systemd] => (item=3.11)
changed: [debian13-systemd] => (item=3.11)
```

and then the following for the files in their respective containers:

```console
$ docker container exec debian12-systemd ls -l /usr/lib/python3.11/EXTERNALLY-MANAGED
-rw-r--r-- 1 root root 645 May  7 19:45 /usr/lib/python3.11/EXTERNALLY-MANAGED
$ docker container exec debian13-systemd ls -l /usr/lib/python3.11/EXTERNALLY-MANAGED
-rw-r--r-- 1 root root 645 Apr 10 13:16 /usr/lib/python3.11/EXTERNALLY-MANAGED
```

So I believe what is happening is the `python3`/`python3.11` package is being upgraded (and the file is restored as part of that) so the file exists before we come in to ensure it exists. This makes sense because Debian 13 is the current Testing release and packages will fluctuate more compared to Debian 12 which is fairly static. I decided to fully verify this by doing the following:

```console
$ docker run --name test-container --detach --privileged --volume /sys/fs/cgroup:/sys/fs/cgroup:rw --cgroupns=host cisagov/docker-debian13-ansible:latest
0ba1bbc9eb551810b33a8afac2ef91398c5153bbbbabbe679b7ce81dc102c92a
$ docker exec -it test-container /bin/bash
root@0ba1bbc9eb55:/# ls -l /usr/lib/python3.11/EXTERNALLY-MANAGED
ls: cannot access '/usr/lib/python3.11/EXTERNALLY-MANAGED': No such file or directory
root@0ba1bbc9eb55:/# dpkg --search /usr/lib/python3.11/EXTERNALLY-MANAGED
libpython3.11-stdlib:amd64: /usr/lib/python3.11/EXTERNALLY-MANAGED
root@0ba1bbc9eb55:/# apt update
<omitted>
root@0ba1bbc9eb55:/# apt full-upgrade --yes
<omitted>
Preparing to unpack .../3-libpython3.11-stdlib_3.11.9-1_amd64.deb ...
Unpacking libpython3.11-stdlib:amd64 (3.11.9-1) over (3.11.8-1) ...
<omitted>
Setting up libpython3-stdlib:amd64 (3.11.8-1) ...
<omitted>
root@0ba1bbc9eb55:/# ls -l /usr/lib/python3.11/EXTERNALLY-MANAGED
-rw-r--r-- 1 root root 645 Apr 10 13:16 /usr/lib/python3.11/EXTERNALLY-MANAGED
```

This shows that the file is restored when performing a `full-upgrade` (as is done in [cisagov/ansible-role-upgrade](https://github.com/cisagov/ansible-role-upgrade/blob/29c7dafe9e43604eaea6765b6d4d7b0777cf2433/tasks/apt.yml#L8-L10)).

Reply:
> What do you think about changing the dummy Molecule test to check that the system python is externally managed? That would seem to be a worthwhile thing to do here.

What about this part of my comment?  I think this would be a worthwhile thing to do.

Reply:
@jsf9k I'm fine with doing that but is the test useful long-term? Ideally we will start using our own SystemD-enabled Docker images and we will be able to revert the work in this PR as the configurations should not be doing any destructive changes. I would greatly like to add some more "meat" to this role so that there is more framework in place for potential consumers outside of our team.